geospaas_harvesting/crawlers.py

```diff
@@ -219,9 +221,26 @@ def _explore_page(self, folder_url):
                     resource_url = f"{current_location}/{link}"
                     if resource_url not in self._urls:
                         if self._intersects_time_range(*(self._dataset_timestamp(link),) * 2):
-                            self.LOGGER.debug("Adding '%s' to the list of resources.", link)
+                            self.LOGGER.debug("Adding downloadable form of '%s' to the list of resources.", link)
+                            resource_url = self.get_download_url(resource_url)
                             self._urls.append(resource_url)
 
+    def get_download_url(self, resource_url):
+        """
+        This method should return the downloadable form of the crawled link. It means providing the
+        proper link so that the user in the datasetURI box is able to click on it and immediately
+        download the file conveniently.
```

Comment:
We are not in the context of the UI here
```suggestion
        This method should return the downloadable form of the crawled link. It means providing a
        direct download link.
```

Reply:
Done in the next commit